test/cases/generate-critical-css/generate-critical-css.spec.js

```diff
@@ -10,49 +10,74 @@ describe('HtmlCriticalWebpackPlugin Cases: Generate Critical CSS', () => {
-        const indexHtmlDom = new JSDOM(indexHtmlString);
-        const linkTags = indexHtmlDom.window.document.querySelectorAll('link');
-        
-        assert.equal(linkTags.length, 2);
-        assert.equal(linkTags[0].getAttribute('rel'), 'preload');
-        assert.equal(linkTags[0].getAttribute('as'), 'style');
+    it('should generate the expected critical inline <style> tag', () => {
+      const inlineStyleTags = document.querySelectorAll('style');
 
-        assert.equal(linkTags[1].getAttribute('rel'), 'preload');
-        assert.equal(linkTags[1].getAttribute('as'), 'style');
+      assert.equal(inlineStyleTags.length, 1);
+      assert.equal(inlineStyleTags[0].getAttribute('type'), 'text/css');
+    });
 
-        done();
-      });
+    it('should generate the expected critical <link> tag', () => {
+      const linkTags = document.querySelectorAll('link');
+      
+      assert.equal(linkTags.length, 2);
+      assert.equal(linkTags[0].getAttribute('rel'), 'preload');
+      assert.equal(linkTags[0].getAttribute('as'), 'style');
+
+      assert.equal(linkTags[1].getAttribute('rel'), 'preload');
+      assert.equal(linkTags[1].getAttribute('as'), 'style');
     });
 
-    it('should generate the expected critical <noscript> tag', (done) => {
-      webpack(webpackConfig, () => {
+    it('should generate the expected critical <noscript> tag', () => {
+      const noscriptTags = document.querySelectorAll('noscript');
+      
+      assert.equal(noscriptTags.length, 2);
+    });
+
+  });
+
+  describe('public path', () => {
+    let document;
+
+    const output = Object.assign({ publicPath: '//cdn.example.com/' }, webpackConfig.output);
+    const config = Object.assign({}, webpackConfig, { output: output });
+
+    before((done) => {
+      webpack(config, () => {
         const indexHtmlString = fs.readFileSync(`${buildDirectory}/index.html`, { encoding: 'utf8' });
-        const indexHtmlDom = new JSDOM(indexHtmlString);
-        const noscriptTags = indexHtmlDom.window.document.querySelectorAll('noscript');
-        
-        assert.equal(noscriptTags.length, 2);
 
+        document = (new JSDOM(indexHtmlString)).window.document;
+        
         done();
       });
     });
 
+    it('should generate the expected critical inline <style> tag', () => {
+      const inlineStyleTags = document.querySelectorAll('style');
+
+      assert.equal(inlineStyleTags.length, 1);
+      assert.equal(inlineStyleTags[0].getAttribute('type'), 'text/css');
+    });
+
+    it('assets reference should start with public path', () => {
+      const cssAssets = [].slice.call(document.querySelectorAll('link[href]'), 0).map(link => link.getAttribute('href'));
+      const jsAssets = [].slice.call(document.querySelectorAll('script[src]'), 0).map(script => script.getAttribute('src'));
+      
+      assert.ok(cssAssets.concat(jsAssets).every(asset => (/^\/\/cdn\.example\.com/).test(asset)));
```

Comment:
awesome stuff @lili21 !  

I have two small observations here, take them as you will 😊 

1.  I would create two `it` blocks here, one for CSS assets, and one for JS assets (and update the titles of them as well to reflect their intent).  In this way, if for some reason only the JS or CSS assets fail the spec, it would only fail the related CSS or JS spec and would provide immediate insight where to start, rather than having to guess which one.  You never know, and also it will keep each spec a little slimmer / more focused, which I think is a good practice

2.  I generally favor making sure the expected in an `assert` comes from the same input source as provided to the unit under test.  This makes the test less brittle if the value of the input ever changes.  So in this case, asserting on `config.publicPath`,  e.g.
```javascript
assert.ok(cssAssets.every(asset => (/^`${config.publicPath}`/).test(asset)));  // I'm guessing on the syntax here, but you get the idea
```